include/boost/math/distributions/non_central_beta.hpp

```diff
@@ -86,7 +86,12 @@ namespace boost
                }
                pois *= i / l2;
                beta += xterm;
-               xterm *= (a + i - 1) / (x * (a + b + i - 2));
+
+               if (a + b + i != 2)
```

Comment:
Rather than putting an `if` inside the inner loop, it would be better to move calculation of xterm to the start of the loop: albeit it's inconvenient, and we have to be careful to get the algebra right ;)

Again this should wait till after the next release.